impl/src/main/java/org/ehcache/loaderwriter/writebehind/WriteBehindConfig.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.loaderwriter.writebehind;
+
+import org.ehcache.spi.loaderwriter.CacheLoaderWriterConfiguration;
+
+/**
+ * @author Abhilash
+ *
+ */
+public class WriteBehindConfig {
+  
+  /**
+   * Default minimum write delay
+   */
+  public static final int DEFAULT_MIN_WRITE_DELAY = 1;
+  /**
+   * Default maximum write delay
+   */
+  public static final int DEFAULT_MAX_WRITE_DELAY = 1;
+  /**
+   * Default rate limit per second
+   */
+  public static final int DEFAULT_RATE_LIMIT_PER_SECOND = 0;
+  /**
+   * Default write coalescing behavior
+   */
+  public static final boolean DEFAULT_WRITE_COALESCING = false;
+  /**
+   * Default writeBatching behavior
+   */
+  public static final boolean DEFAULT_WRITE_BATCHING = false;
+  /**
+   * Default write batch size
+   */
+  public static final int DEFAULT_WRITE_BATCH_SIZE = 1;
+  /**
+   * Default retry attempts
+   */
+  public static final int DEFAULT_RETRY_ATTEMPTS = 0;
+  /**
+   * Default retry attempt delay
+   */
+  public static final int DEFAULT_RETRY_ATTEMPT_DELAY_SECONDS = 1;
+
+  /**
+   * Default concurrency level for write behind
+   */
+  public static final int DEFAULT_WRITE_BEHIND_CONCURRENCY = 1;
+
+  /**
+   * Default max queue size for write behind
+   */
+  public static final int DEFAULT_WRITE_BEHIND_MAX_QUEUE_SIZE = 0;
+
+  private boolean iswriteBehind = false;
+  private int minWriteDelay = DEFAULT_MIN_WRITE_DELAY;
+  private int maxWriteDelay = DEFAULT_MAX_WRITE_DELAY;
+  private int rateLimitPerSecond = DEFAULT_RATE_LIMIT_PER_SECOND;
+  private boolean writeCoalescing = DEFAULT_WRITE_COALESCING;
+  private boolean writeBatching = DEFAULT_WRITE_BATCHING;
+  private int writeBatchSize = DEFAULT_WRITE_BATCH_SIZE;
+  private int retryAttempts = DEFAULT_RETRY_ATTEMPTS;
+  private int retryAttemptDelaySeconds = DEFAULT_RETRY_ATTEMPT_DELAY_SECONDS;
+  private int writeBehindConcurrency = DEFAULT_WRITE_BEHIND_CONCURRENCY;
+  private int writeBehindMaxQueueSize = DEFAULT_WRITE_BEHIND_MAX_QUEUE_SIZE;
+  
+  private String alias;
+  
+  public WriteBehindConfig getWriteBehindConfig(CacheLoaderWriterConfiguration cacheLoaderWriterConfiguration) {
+    WriteBehindConfig config = new WriteBehindConfig();
+    // TODO get Alias
+    config.setAlias("");
```

Comment:
Alias is just for logging purpose , may be we can ignore it and just log the cacheloaderwriter class instead